io/EptAddonWriter.cpp

```diff
@@ -0,0 +1,252 @@
+    for (const std::string path : m_addonsArg->getMemberNames())
+    {
+        const auto endpoint(
+                m_arbiter->getEndpoint(arbiter::fs::expandTilde(path)));
+        const std::string dimName((*m_addonsArg)[path].asString());
+        const Dimension::Id id(layout.findDim(dimName));
+        m_addons.emplace_back(new Addon(layout, endpoint, id));
+    }
+}
+
+void EptAddonWriter::write(const PointViewPtr view)
+{
+    for (const auto& addon : m_addons)
+    {
+        log()->get(LogLevel::Debug) << "Writing addon dimension " <<
+            addon->name() << " to " << addon->ep().prefixedRoot() << std::endl;
+
+        writeOne(view, *addon);
+
+        log()->get(LogLevel::Debug) << "\tWritten" << std::endl;
+    }
+}
+
+void EptAddonWriter::writeOne(const PointViewPtr view, const Addon& addon) const
+{
+    std::vector<std::vector<char>> buffers;
+    buffers.reserve(m_hierarchy.size());
+
+    // Create an addon buffer for each node we're going to write.
+    for (const auto& p : m_hierarchy)
+    {
+        buffers.emplace_back(p.second * addon.size(), 0);
+    }
+
+    // Fill in our buffers with the data from the view.
+    PointRef pr(*view);
+    uint64_t nodeId(0);
+    uint64_t pointId(0);
+    for (uint64_t i(0); i < view->size(); ++i)
+    {
+        pr.setPointId(i);
+        nodeId = pr.getFieldAs<uint64_t>(m_nodeIdDim);
+
+        // Node IDs are 1-based to distinguish points that do not come from the
+        // EPT reader.
+        if (!nodeId) continue;
+
+        nodeId -= 1;
+        pointId = pr.getFieldAs<uint64_t>(m_pointIdDim);
+
+        auto& buffer(buffers.at(nodeId));
+        assert(pointId * addon.size() + addon.size() <= buffer.size());
+        char* dst = buffer.data() + pointId * addon.size();
+        pr.getField(dst, addon.id(), addon.type());
+    }
+
+    const arbiter::Endpoint& ep(addon.ep());
+    const arbiter::Endpoint dataEp(ep.getSubEndpoint("ept-data"));
+    const arbiter::Endpoint hierEp(ep.getSubEndpoint("ept-hierarchy"));
+
+    if (ep.isLocal())
+    {
+        arbiter::fs::mkdirp(dataEp.root());
+        arbiter::fs::mkdirp(hierEp.root());
+    }
+
+    // Write the binary dimension data for the addon.
+    nodeId = 0;
+    for (const auto& p : m_hierarchy)
+    {
+        const Key key(p.first);
+
+        m_pool->add([&dataEp, &buffers, key, nodeId]()
+        {
+            dataEp.put(key.toString() + ".bin", buffers.at(nodeId));
+        });
+
+        ++nodeId;
+    }
+
+    m_pool->await();
+
+    // Write the addon hierarchy data.
+    Json::Value h;
+    const Key key(m_info->bounds());
+    writeHierarchy(h, Key(m_info->bounds()), hierEp);
+    hierEp.put(key.toString() + ".json", h.toStyledString());
+
+    m_pool->await();
+
+    // Write the top-level addon metadata.
+    Json::Value meta;
+    meta["type"] = getTypeString(addon.type());
+    meta["size"] = static_cast<Json::UInt64>(addon.size());
+    meta["version"] = "1.0.0";
+    meta["dataType"] = "binary";
```

Comment:
I'm not very familiar with the EPT spec at this point. Is there another option?

Reply:
For EPT itself (rather than addons), `laszip` is the other supported option, which isn't appropriate for single-attribute addons.  I'll shortly be adding Draco compression support which will be an option for both EPT and for addons.  For now, the answer to your question is no - there is not another option.